src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs

```diff
@@ -54,19 +55,69 @@ public static (Type middleware, Type options) LookupMiddleware(Type endpointType
         /// <exception cref="InvalidOperationException">When T is not found in service container</exception>
-        public static void Map<TEndpoint>(this IEndpointRouteBuilder endpoints)
+        public static IEndpointConventionBuilder Map<TEndpoint>(this IEndpointRouteBuilder endpoints, EndpointCollectionConventionBuilder conventionBuilder = null)
         where TEndpoint : IEndpoint
+        {
+            return MapActuatorEndpoint(endpoints, typeof(TEndpoint), conventionBuilder);
+        }
+
+        public static IEndpointConventionBuilder MapAllActuators(this IEndpointRouteBuilder endpoints, MediaTypeVersion version = MediaTypeVersion.V2)
+        {
+            var conventionBuilder = new EndpointCollectionConventionBuilder();
+            endpoints.Map<ActuatorEndpoint>(conventionBuilder);
+
+            if (Platform.IsWindows)
+            {
+                if (version == MediaTypeVersion.V2)
+                {
+                    endpoints.Map<ThreadDumpEndpoint_v2>(conventionBuilder);
+                }
+                else
+                {
+                    endpoints.Map<ThreadDumpEndpoint>(conventionBuilder);
+                }
+            }
+
+            if (Platform.IsWindows || Platform.IsLinux)
+            {
+                endpoints.Map<HeapDumpEndpoint>(conventionBuilder);
+            }
+
+            endpoints.Map<EnvEndpoint>(conventionBuilder);
+            endpoints.Map<RefreshEndpoint>(conventionBuilder);
+            endpoints.Map<InfoEndpoint>(conventionBuilder);
+            endpoints.Map<HealthEndpoint>(conventionBuilder);
+            endpoints.Map<LoggersEndpoint>(conventionBuilder);
+            if (version == MediaTypeVersion.V2)
+            {
+                endpoints.Map<HttpTraceEndpoint>(conventionBuilder);
+            }
+            else
+            {
+                endpoints.Map<TraceEndpoint>(conventionBuilder);
+            }
+
+            endpoints.Map<MappingsEndpoint>(conventionBuilder);
+            endpoints.Map<MetricsEndpoint>(conventionBuilder);
+            endpoints.Map<PrometheusScraperEndpoint>(conventionBuilder);
+
+            return conventionBuilder;
+        }
+
+        internal static IEndpointConventionBuilder MapActuatorEndpoint(this IEndpointRouteBuilder endpoints, Type typeEndpoint, EndpointCollectionConventionBuilder conventionBuilder = null)
         {
             if (endpoints == null)
             {
                 throw new ArgumentNullException(nameof(endpoints));
             }
 
-            var (middleware, optionsType) = LookupMiddleware(typeof(TEndpoint));
+            var (middleware, optionsType) = LookupMiddleware(typeEndpoint);
             var options = endpoints.ServiceProvider.GetService(optionsType) as IEndpointOptions;
             var mgmtOptionsCollection = endpoints.ServiceProvider.GetServices<IManagementOptions>();
+            var builder = conventionBuilder ?? new EndpointCollectionConventionBuilder();
```

Comment:
it might be clearer to do `conventionBuilder ??= new EndpointCollectionConventionBuilder();` here